src/Sulu/Component/Content/Document/Subscriber/RouteSubscriber.php

```diff
@@ -38,10 +81,77 @@ public function handleMetadataLoad(MetadataLoadEvent $event)
             return;
         }
 
-        $metadata->addFieldMapping('targetDocument', [
-            'encoding' => 'system',
-            'property' => self::DOCUMENT_TARGET_FIELD,
-            'type' => 'reference',
-        ]);
+        $metadata->addFieldMapping(
+            'created',
+            [
+                'encoding' => 'system',
+                'property' => self::CREATED,
+                'type' => 'date',
+            ]
+        );
+    }
+
+    /**
+     * Updates the route for the given document and creates history routes if necessary.
+     *
+     * @param PersistEvent $event
+     */
+    public function handlePersist(PersistEvent $event)
+    {
+        $document = $event->getDocument();
+
+        if (!$document instanceof RouteBehavior) {
+            return;
+        }
+
+        $node = $event->getNode();
+
+        $targetDocument = $document->getTargetDocument();
+
+        if (!$targetDocument instanceof WebspaceBehavior || !$targetDocument instanceof ResourceSegmentBehavior) {
+            return;
+        }
+
+        $targetNode = $this->documentInspector->getNode($targetDocument);
+
+        // copy new route to old position
+        $webspaceKey = $targetDocument->getWebspaceName();
+        $locale = $this->documentInspector->getLocale($targetDocument);
+
+        $routePath = $this->sessionManager->getRoutePath($webspaceKey, $locale, null)
+            . $targetDocument->getResourceSegment();
+
+        // create a route node if it is not a new document and the path changed
+        $documentPath = $this->documentInspector->getPath($document);
+        if ($documentPath && $documentPath != $routePath) {
+            /** @var RouteDocument $newRouteDocument */
+            $newRouteDocument = $this->documentManager->create('route');
+            $newRouteDocument->setTargetDocument($targetDocument);
+            $this->documentManager->persist($newRouteDocument, $locale, [
+                'path' => $routePath,
+                'auto_create' => true,
+            ]);
+
+            // change routes in old position to history
+            $this->changeOldPathToHistoryRoutes($node, $this->documentInspector->getNode($newRouteDocument));
+        } else {
+            $node->setProperty('sulu:content', $targetNode);
+        }
+    }
+
+    /**
+     * Changes the old route to a history route and redirect to the new route.
+     *
+     * @param NodeInterface $oldRouteNode
+     * @param NodeInterface $newRouteNode
+     */
+    private function changeOldPathToHistoryRoutes(NodeInterface $oldRouteNode, NodeInterface $newRouteNode)
+    {
+        $oldRouteNode->setProperty('sulu:content', $newRouteNode);
```

Comment:
This is a very bad thing to do as I think we already know -- imagine that you change the RL 200 times. Then there will be 200 redirects.

The content should be the content not the route.